npm/parse.go

```diff
@@ -394,177 +494,351 @@ func parseEgress(ns string, targetSets []string, rules []networkingv1.NetworkPol
 			continue
 		}
 
-		if !portRuleExists && !toRuleExists {
-			allow := &iptm.IptEntry{
-				Name:       targetSet,
-				HashedName: hashedTargetSetName,
-				Chain:      util.IptablesAzureEgressPortChain,
-				Specs: []string{
-					util.IptablesMatchFlag,
-					util.IptablesSetFlag,
-					util.IptablesMatchSetFlag,
-					hashedTargetSetName,
-					util.IptablesSrcFlag,
-					util.IptablesJumpFlag,
-					util.IptablesAccept,
-				},
-			}
-			entries = append(entries, allow)
-			continue
+		// allow kube-system
+		hashedKubeSystemSet := util.GetHashedName(util.KubeSystemFlag)
+		allowKubeSystemEgress := &iptm.IptEntry{
+			Name:       util.KubeSystemFlag,
+			HashedName: hashedKubeSystemSet,
+			Chain:      util.IptablesAzureEgressPortChain,
+			Specs: []string{
+				util.IptablesMatchFlag,
+				util.IptablesSetFlag,
+				util.IptablesMatchSetFlag,
+				hashedTargetSetName,
+				util.IptablesSrcFlag,
+				util.IptablesMatchFlag,
+				util.IptablesSetFlag,
+				util.IptablesMatchSetFlag,
+				hashedKubeSystemSet,
+				util.IptablesDstFlag,
+				util.IptablesJumpFlag,
+				util.IptablesAccept,
+			},
 		}
+		entries = append(entries, allowKubeSystemEgress)
 
-		if !portRuleExists {
-			entry := &iptm.IptEntry{
-				Name:       targetSet,
-				HashedName: hashedTargetSetName,
-				Chain:      util.IptablesAzureEgressPortChain,
-				Specs: []string{
-					util.IptablesMatchFlag,
-					util.IptablesSetFlag,
-					util.IptablesMatchSetFlag,
-					hashedTargetSetName,
-					util.IptablesDstFlag,
-					util.IptablesJumpFlag,
-					util.IptablesAzureEgressToChain,
-				},
+		for _, rule := range rules {
+			for _, portRule := range rule.Ports {
+				protPortPairSlice = append(protPortPairSlice,
+					&portsInfo{
+						protocol: string(*portRule.Protocol),
+						port:     fmt.Sprint(portRule.Port.IntVal),
+					})
+
+				portRuleExists = true
 			}
-			entries = append(entries, entry)
-		} else {
-			for _, protPortPair := range protPortPairSlice {
-				entry := &iptm.IptEntry{
+
+			if rule.To != nil {
+				for _, toRule := range rule.To {
+					if toRule.PodSelector != nil {
```

Comment:
you can OR these in if statement

Reply:
That will make the line too long, especially if more fields will be introduced in the future..